snapcraft/cli/remote.py

```diff
@@ -124,6 +111,9 @@ def remote_build(
 
     project = get_project()
 
+    if project.info.base not in ("core", "core16", "core18"):
+        raise errors.InvalidBaseError(base=project.info.base)
```

Comment:
Maybe "requires" base?
Can you quickly try if the builder errors out appropriately if you try and do something like "base: core20"?

Reply:
Add a TODO to use Project.is_legacy, I will work on that.

Reply:
How about for an error: 
```
class UnsupportedBaseError(RemoteBuildBaseError):

    fmt = (
        "Remote build currently requires that the project uses bases.\n"
        "Please specify an appropriate `base` keyword and try again."
    )
```

Reply:
`base` is really unprocessed with project.info.  So whatever I put into base is considered valid here (e.g. core20, core21) until the builder itself determines it isn't valid (which you'll see with the other PR #2642).

I updated to use get_build_base() for the time being, I think that's most appropriate?